### PR TITLE
DIV-6760,6758,6824: respondent solicitor

### PR DESCRIPTION
**Pull request by madjava**
# Description
https://tools.hmcts.net/jira/browse/DIV-6758
https://tools.hmcts.net/jira/browse/DIV-6760
https://tools.hmcts.net/jira/browse/DIV-6824

**Deployement link** - https://div-pfe-pr-821.service.core-compute-preview.internal/

Pull request includes both DIV-6758 and DIV-6760

Screenshot 2021-03-10 at 15 02 48

Screenshot 2021-03-10 at 15 02 56

Screenshot 2021-03-10 at 15 03 07

### CODE REVIEW

File: app/steps/respondent/correspondence/solicitor-search/template.html [hunk 1]
```diff
@@ -30,53 +30,44 @@
 
     {# Solicitors firms search result section - Items found #}
     <div class="govuk-form-group">
-      <label class="govuk-label--s govuk-label">
-        Results
-      </label>
-      <main class="govuk-main-wrapper govuk-!-padding-top-1 govuk-!-padding-bottom-0">
```
Comment (Jamesnelson117) [inline app/steps/respondent/correspondence/solicitor-search/template.html]: @madjava another FYI - the page already is already wrapped in a `main`. Having another one isnt great for semantics

File: app/steps/respondent/correspondence/solicitor-search/template.html [hunk 2]
```diff
@@ -30,53 +30,44 @@
 
     {# Solicitors firms search result section - Items found #}
     <div class="govuk-form-group">
-      <label class="govuk-label--s govuk-label">
```
Comment (Jamesnelson117) [inline app/steps/respondent/correspondence/solicitor-search/template.html]: @madjava FYI - label should only really be used within a <form> element to identify and link to an input element.

File: app/services/organisationService.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+   */
+  getOrganisationByName: (status, organisationName) => {
+    return client.getOrganisationByName(status, organisationName);
+  }
+};
+
+module.exports = {
+  setup: (auth, serviceAuth) => {
+    // Use the mock client if the microservice key is not set.
+    client = hasMandatoryTokens(auth, serviceAuth) ? new OrganisationClient(auth, serviceAuth) : mockedClient;
```
[the remark this answers is not in the document]
Comment (yarpo) [reply app/services/organisationService.js:25]: In production this should be a failure when we don't have valid tokens

File: app/steps/respondent/correspondence/solicitor-search/partials/checkYourAnswersTemplate.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<tr class="govuk-table__row">
+  <th class="govuk-table__cell" scope="row">{{ content.question | safe }}</th>
+  <td class="govuk-table__cell">
+    {%- if fields.respondentSolicitorOrganisation -%}
```
Comment (yarpo) [inline app/steps/respondent/correspondence/solicitor-search/partials/checkYourAnswersTemplate.html:4]: does this:

```
 {%-
```

mean it's commented out?

**Review by yarpo — APPROVED** [review]

**Review by yarpo — APPROVED** [review]

**Review by Jamesnelson117 — APPROVED** [review]
LGTM

File: app/views/common/components/respondentSolicitorComponents.html [hunk 5]
```diff
@@ -39,6 +39,25 @@
 </div>
 {% endmacro %}
 
+{#
+organisationDetail
+organisation - (required) An organisation object
+#}
+{% macro organisationDetail(organisation) %}
```
Comment (Jamesnelson117) [inline app/views/common/components/respondentSolicitorComponents.html:46]: Nice

**Comment by madjava** [comment]
Just to see if the earlier available env is still active

**Comment by yarpo** [comment]
duplicated:
https://github.com/hmcts/div-petitioner-frontend/pull/837

**Pull request merged**